instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/properties/SpringConfigProperties.java

```diff
@@ -117,29 +118,45 @@ public Duration getDuration(String name) {
   @SuppressWarnings("unchecked")
   @Override
   public Map<String, String> getMap(String name) {
+    Map<String, String> otelSdkMap = otelSdkProperties.getMap(name);
     // maps from config properties are not supported by Environment, so we have to fake it
     switch (name) {
       case "otel.resource.attributes":
-        return resourceProperties.getAttributes();
+        return mergeWithOtel(resourceProperties.getAttributes(), otelSdkMap);
       case "otel.exporter.otlp.headers":
-        return otlpExporterProperties.getHeaders();
+        return mergeWithOtel(otlpExporterProperties.getHeaders(), otelSdkMap);
       case "otel.exporter.otlp.logs.headers":
-        return otlpExporterProperties.getLogs().getHeaders();
+        return mergeWithOtel(otlpExporterProperties.getLogs().getHeaders(), otelSdkMap);
       case "otel.exporter.otlp.metrics.headers":
-        return otlpExporterProperties.getMetrics().getHeaders();
+        return mergeWithOtel(otlpExporterProperties.getMetrics().getHeaders(), otelSdkMap);
       case "otel.exporter.otlp.traces.headers":
-        return otlpExporterProperties.getTraces().getHeaders();
+        return mergeWithOtel(otlpExporterProperties.getTraces().getHeaders(), otelSdkMap);
       default:
         break;
     }
 
     String value = environment.getProperty(name);
     if (value == null) {
-      return fallback.getMap(name);
+      return otelSdkMap;
     }
     return (Map<String, String>) parser.parseExpression(value).getValue();
   }
 
+  /**
+   * If you specify the environment variable <code>OTEL_RESOURCE_ATTRIBUTES_POD_NAME</code>, then
+   * Spring Boot will ignore <code>OTEL_RESOURCE_ATTRIBUTES</code>, which violates the principle of
+   * least surprise. This method merges the two maps, giving precedence to <code>
+   * OTEL_RESOURCE_ATTRIBUTES_POD_NAME</code>, which is more specific and which is also the value
```

Comment:
the env var mapping makes me a little nervous since some otel resource attribute names include underscores

but I think we can argue that this is just a spring configuration feature(?)

Reply:
I can't follow - what are you nervous about? Maybe an example would help.

Reply:
some OTel resource attributes have underscores in them, and so it's not clear to me how these would work with env vars, e.g. does `OTEL_RESOURCE_ATTRIBUTES_X_Y_Z` correspond to resource attribute `x.y.z`, `x.y_z`, `x_y.z`, or `x_y_z`?

Reply:
BTW, `OTEL_RESOURCE_ATTRIBUTES_X_Y_Z` results in `x.y.z` - which is indeed a spring boot feature.

Reply:
`OTEL_RESOURCE_ATTRIBUTES_X_Y_Z` should only be used to update an x.y.z or x_y_z that you have defined in application.yaml

oh, so it will auto-match to the right one if you have it already defined in application.yaml?

Reply:
Yes, it will auto match.